tests/tests/helpers.rs

```diff
@@ -159,14 +183,83 @@ pub async fn server() -> SocketAddr {
 		}
 	}
 
-	module.register_async_method::<Result<(), CustomError>, _, _>("err", |_, _| async { Err(CustomError) }).unwrap();
+	module.register_async_method::<Result<(), CustomError>, _, _>("err", |_, _, _| async { Err(CustomError) }).unwrap();
 
-	let addr = server.local_addr().unwrap();
+	let listener = tokio::net::TcpListener::bind("127.0.0.1:0").await.unwrap();
+	let addr = listener.local_addr().unwrap();
+	let (stop_hdl, server_hdl) = stop_channel();
+	let methods: Methods = module.into();
 
-	let server_handle = server.start(module);
+	let methods2 = methods.clone();
+	tokio::spawn(async move {
```

Comment:
Ah I geuss this is all added so that we can inject connection details!

I wonder whether there's something wecan do in the future to simplify this "low level" initialising

Reply:
We could probably inject the connection id by default but it's quite weird to explain/document it so I didn't do it.

What does @lexnv reckon? :)

Reply:
I think this is fine, substrate could then inject the connection ID in a similar way